.ci.sh

```diff
@@ -39,16 +39,13 @@ run_tests() {
 }
 
 run_tests_with_coverage(){
-  # NB: Coverage does not currently work: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/2702
+  # NB: Testing using example from https://github.com/codecov/example-python
   activate_venv_sct
-  pip install coverage
-  echo -ne "import coverage\ncov = coverage.process_startup()\n" > sitecustomize.py
-  echo -ne "[run]\nconcurrency = multiprocessing\nparallel = True\n" > .coveragerc
-  COVERAGE_PROCESS_START="$PWD/.coveragerc" COVERAGE_FILE="$PWD/.coverage" pytest
-  coverage combine
+  pip install pytest-cov
+  pytest --cov=spinalcordtoolbox --cov-config setup.cfg --cov-branch --cov-report=xml
```

Comment:
One other thing I'd like to try is running the API tests and the CLI tests separately, then having separate coverage reports for each.

That way, we can know if a LOC is tested via a unit test, an integration test, or both. That distinction is pretty important! (I _think_ codecov is smart enough to give a fancy report for this.)

Reply:
Huzzah! I've used codecov's "flags" to give us toggles for coverage from API tests and/or CLI tests:

![Screenshot from 2021-07-14 20-09-27](https://user-images.githubusercontent.com/16181459/125708295-42126396-ac71-4392-bccc-1e2cb4ad2fcc.png)

(Maybe I should name these tags `testing/api` and `testing/cli`, though.)

Reply:
> (Maybe I should name these tags testing/api and testing/cli, though.)

Done in f6412bc.